### ISSUE TITLE
APP常驻后台再打开，出现了崩溃。

### ISSUE THREAD

**Issue by sleep1234** [nextcloud/android#12659]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

长时间保持后台，然后再打开

### Expected behaviour

然后就出现了故障报错

### Actual behaviour

保持后台打开之后突然就这样了。

### Android version

安卓14

### Device brand and model

小米14Pro

### Stock or custom OS?

Stock

### Nextcloud android app version

3.28.0

### Nextcloud server version

28.0.3

### Using a reverse proxy?

No

### Cause of error
```java
Exception in thread "main" java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object java.util.Stack.pop()' on a null object reference
    at com.owncloud.android.ui.activity.FileDisplayActivity.popSortListGroupVisibility(FileDisplayActivity.java:2505)
    at com.owncloud.android.ui.activity.FileDisplayActivity.popBack(FileDisplayActivity.java:1048)
    at com.owncloud.android.ui.activity.FileDisplayActivity.onBackPressed(FileDisplayActivity.java:1003)
    at android.app.Activity.onKeyUp(Activity.java:4174)
    at android.view.KeyEvent.dispatch(KeyEvent.java:2958)
    at android.app.Activity.dispatchKeyEvent(Activity.java:4547)
    at androidx.core.app.ComponentActivity.superDispatchKeyEvent(ComponentActivity.java:120)
    at androidx.core.view.KeyEventDispatcher.dispatchKeyEvent(KeyEventDispatcher.java:85)
    at androidx.core.app.ComponentActivity.dispatchKeyEvent(ComponentActivity.java:138)
    at androidx.appcompat.app.AppCompatActivity.dispatchKeyEvent(AppCompatActivity.java:604)
    at androidx.appcompat.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:60)
    at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.dispatchKeyEvent(AppCompatDelegateImpl.java:3413)
    at com.android.internal.policy.DecorView.dispatchKeyEvent(DecorView.java:424)
    at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:7696)
    at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:7544)
    at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6935)
    at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6992)
    at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6958)
    at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:7123)
    at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:6966)
    at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:7180)
    at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6939)
    at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6992)
    at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6958)
    at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:6966)
    at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6939)
    at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6992)
    at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6958)
    at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:7156)
    at android.view.ViewRootImpl$ImeInputStage.onFinishedInputEvent(ViewRootImpl.java:7384)
    at android.view.inputmethod.InputMethodManager$PendingEvent.run(InputMethodManager.java:4355)
    at android.view.inputmethod.InputMethodManager.invokeFinishedInputEventCallback(InputMethodManager.java:3757)
    at android.view.inputmethod.InputMethodManager.finishedInputEvent(InputMethodManager.java:3748)
    at android.view.inputmethod.InputMethodManager.-$$Nest$mfinishedInputEvent(Unknown Source:0)
    at android.view.inputmethod.InputMethodManager$ImeInputEventSender.onInputEventFinished(InputMethodManager.java:4332)
    at android.view.InputEventSender.dispatchInputEventFinished(InputEventSender.java:154)
    at android.os.MessageQueue.nativePollOnce(Native Method)
    at android.os.MessageQueue.next(MessageQueue.java:341)
    at android.os.Looper.loopOnce(Looper.java:176)
    at android.os.Looper.loop(Looper.java:318)
    at android.app.ActivityThread.main(ActivityThread.java:8669)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:561)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30280090`
* Build flavor: `gplay`

### Device information
* Brand: `Xiaomi`
* Device: `shennong`
* Model: `23116PN5BC`
* Id: `UKQ1.230804.001`
* Product: `shennong`

### Firmware
* SDK: `34`
* Release: `14`
* Incremental: `V816.0.24.2.19.DEV`

**Comment by ZetaTom** [comment]
Thank you for reporting this issue. Please note that issues and bug reports should generally be submitted in English.

Guessing from the provided log, the problem seems to originate from `previousSortGroupState`. As this was fixed (#12540) and subsequently removed in #12584, it should be working on master.

**Comment by ZetaTom** [comment]
Duplicate of #12649; should be fixed by #12540 and #12584.
